### PR TITLE
Fulltext tweaks

### PR DESCRIPTION
**Pull request by jasonLaster**
* Persists the search query
* Re focuses the input on cmd+shift+f
* Emphasize full text search filename results

Screen Shot 2021-12-17 at 8 59 33 PM

### CODE REVIEW

File: src/devtools/client/debugger/src/components/SecondaryPanes/EventListeners.js [hunk 1]
```diff
@@ -295,16 +295,21 @@ class EventListeners extends Component {
       return null;
     }
 
+    const isHot = points.length > prefs.maxHitsEditable;
+    const title = isHot ? `Cannot view ${event.name} events` : `View ${event.name} events`;
+    console.log(isHot, points.length);
```
Comment (jaril) [inline src/devtools/client/debugger/src/components/SecondaryPanes/EventListeners.js:300]: 🧂

File: src/devtools/client/debugger/src/reducers/ui.js [hunk 2]
```diff
@@ -14,6 +14,8 @@ import { prefs, features } from "../utils/prefs";
 export const createUIState = () => ({
   selectedPrimaryPaneTab: "sources",
   activeSearch: null,
+  fullTextQuery: "",
```
Comment (jaril) [inline src/devtools/client/debugger/src/reducers/ui.js:17]: Nit but probably best to just use the full name here so `fullTextSearchQuery` and `fullTextSearchFocus` so searching for references is easier.

File: src/devtools/client/debugger/src/components/SecondaryPanes/EventListeners.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ import { actions } from "ui/actions";
 import { selectors } from "ui/reducers";
 
 import AccessibleImage from "../shared/AccessibleImage";
-import { features } from "ui/utils/prefs";
+import { features, prefs } from "ui/utils/prefs";
```
Comment (jaril) [inline src/devtools/client/debugger/src/components/SecondaryPanes/EventListeners.js:15]: This commit can probably be moved into its own PR.

File: src/devtools/client/debugger/src/actions/ui.js [hunk 4]
```diff
@@ -219,3 +219,14 @@ export function refreshCodeMirror() {
...
 }
+
+export function setFullTextQuery(query) {
+  return { type: "SET_FULLTEXT_SEARCH", query };
+}
+
+export function focusFullTextInput() {
+  return ({ dispatch }) => {
+    dispatch({ type: "FOCUS_FULLTEXT_SEARCH", focus: true });
+    setTimeout(() => dispatch({ type: "FOCUS_FULLTEXT_SEARCH", focus: false }), 100);
```
Comment (jaril) [inline src/devtools/client/debugger/src/actions/ui.js:230]: What's this doing??

Comment (jcmorrow) [reply src/devtools/client/debugger/src/actions/ui.js:230]: Seems like maybe this would be better in the `onFocus` of the text input? If what you're trying to do is reset it?

File: src/devtools/client/debugger/src/components/FullTextSearch/index.js [hunk 5]
```diff
@@ -69,7 +68,7 @@ export class FullTextSearch extends Component {
       return;
     }
 
-    const query = sanitizeQuery(e.target.value);
+    const sanitizedQuery = sanitizeQuery(e.target.value);
```
Comment (jcmorrow) [inline src/devtools/client/debugger/src/components/FullTextSearch/index.js:71]: Is this an important sanitization? If so we should probably do it on the backend.

**Review by jcmorrow — COMMENTED** [review]
Couple of questions

File: src/devtools/client/debugger/src/actions/ui.js (hunk 4 above)
Comment (jasonLaster) [reply src/devtools/client/debugger/src/actions/ui.js:230]: clever.

File: src/devtools/client/debugger/src/components/FullTextSearch/index.js (hunk 5 above)
Comment (jasonLaster) [reply src/devtools/client/debugger/src/components/FullTextSearch/index.js:71]: that's fair
